.github/labels.md

```diff
@@ -0,0 +1,234 @@
+
+#### Label category `component`
+
+The component lists the parts of the project this Issue affects or is affected
+by.  This should help developers to identify if they have required knowledge or
+interests to handle the issue.  There should be at least one component listed
+for an Issue.
+
+#### Label category `status`
+
+With the status category developers, managers or issue reporters can check if
+an issue needs further interaction and who should carry out this interaction.
+Issue have exactly one status.
+
+
+## Labels
+
+This section lists when an label should be applied and what actions may be taken
+additionally.
+
+
+#### Label `category:bug`
+
+The Issue describes or solves a percieved malfunction within the game.
+
+#### Label `category:feature`
+
+The Pull Request contains a new game feature.  Features are parts of the game
+which a player can interact with e.g.: A new species, a new gameplay mechanic
+or a new command line switch for the server that changes gameplay behaviour.
+
+#### Label `category:idea`
+
+??? No idea, overlaps with `category:feature` or `state:work in progress` IMO.
+
+#### Label `category:question`
+
+??? Questions don't belong into the issue tracker IMO.
+
+#### Label `category:refactoring`
+
+The Pull request contains a new or improved implementation of an existing
+feature or any other code restructuring that doesn't modify the current game
+mechanics or representation.
+
+#### Label `category:request`
+
+??? Questions don't belong into the issue tracker IMO.
+
+#### Label `category:tweak`
+
+??? No idea.
+
+#### Label `component:AI`
+
+The Issue requires knowledge about the python AI decision making code or affects
+it.
+
+#### Label `component:art music text assets`
+
+The Issue requires knowledge about creating game assets or verification if the
+assets match the style goals of the game.
+
+#### Label `component:build system`
+
+The Issue requires knowledge about CMake, Visual Studio, Xcode or the general
+build process.
+
+#### Label `component:common`
+
+??? No idea.
+
+#### Label `component:deployment`
+
+The Issue requires knowledge about the deployment process or the software used
+create the installer. ??? I see an overlap with `component:build system` here.
+
+#### Label `component:documentation`
+
+The Issue requires knowledge about Doxygen or primarly contains documentation.
+
+#### Label `component:game mechanic`
+
+The Pull Request requires knowledge about the currently used or planned game
+mechanics.  The content of the Pull Request may require extensive gameplay
+balance testing.
+
+#### Label `component:infrastructure`
+
+The Pull Request requires knowledge about non-game related infrastructure that
+aids development.  For example this includes issues with the project homepage,
+used external services like continous integration or hosting of release
+binaries.
+
+#### Label `component:internal`
+
+??? No idea.  Thread suggest it's a non-user visible feature, but then it's the
```

Comment:
I don't know if "internal" fits what the "category" labels are intended for. As far as I understand it, the "category" labels define the *kind* of an issue/PR, the "component" labels which *part of the code* is affected (so to speak).

I understand "component:internal" as referring to changes applying to "internal" parts of the code, pretty much like @geoffthemedio defined it above (not networking, UI etc.). I wouldn't know what "category:internal" is supposed to be compared to the other "category" labels.

So keeping that but probably dropping "component:common" sounds reasonable to me. I never really knew when to use that one anyway.